### PR TITLE
Upgraded dependencies on bootstrap branch

### PR DESCRIPTION
**Pull request by colinskow**
- I upgraded all the dependencies as far as I could without breaking anything.
- I also fixed awesome-typescript-loader so it correctly uses `tsconfig.webpack.json` so tree shaking can happen.

Angular v2.3.1 breaks with NgRx on AOT, but this will be fixed when Angular v2.3.2 is released. At this time we can upgrade Angular along with Zone and RxJS to latest versions.

I thoroughly tested everything on the `bootstrap` branch. You should test with Material and Universal before merging to other branches.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -37,7 +37,6 @@
     "curly": false,
     "forin": true,
     "label-position": true,
-    "label-undefined": true,
```
Comment (colinskow) [inline tslint.json]: All these removed rules were deprecated because TypeScript now includes them directly without TSLint.

File: e2e/app.e2e.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { browser } from 'protractor';
```
Comment (colinskow) [inline e2e/app.e2e.ts]: Protractor now has its own typings, so I removed `@types/protractor`

File: package.json [hunk 3]
```diff
@@ -32,11 +32,11 @@
     "test:once": "karma start",
     "test:once:ci": "karma start",
     "test": "karma start",
-    "ci": "npm run e2e && npm run test:once:ci",
+    "ci": "npm run e2e:jit && npm run test:once:ci",
```
Comment (colinskow) [inline package.json]: e2e doesn't build the project, which is important in CI, so I added the jit option

File: package.json [hunk 4]
```diff
@@ -32,11 +32,11 @@
...
     "ci:testall": "npm run e2e:testall && npm run test:once:ci",
     "tslint": "tslint",
     "lint": "npm run tslint \"src/app/**/*.ts\" ",
-    "clean": "npm cache clean && npm run rimraf -- node_modules doc typings coverage dist .awcache dll && npm run clean:compile",
+    "clean": "npm cache clean && npm run clean:compile && npm run rimraf -- node_modules doc typings coverage dist .awcache dll",
```
Comment (colinskow) [inline package.json]: This was in the incorrect order because it deleted the rimraf command before it could complete

File: tsconfig.aot.json [hunk 5]
```diff
@@ -19,10 +19,9 @@
   "exclude": [
     "node_modules",
     "src/compiled",
-    "src/**/*.e2e.ts",
```
Comment (colinskow) [inline tsconfig.aot.json]: I removed this because there are no .e2e files in the src directory. They are in e2e/

**Comment by colinskow** [comment]
I added comments explaining most changes in the `Files Changed` feed.

**Comment by qdouble** [comment]
I'll just merge this and revert anything I see that might be problematic

**Pull request merged**